src/tree/gpu_hist/row_partitioner.cuh

```diff
@@ -125,7 +125,6 @@ class RowPartitioner {
       idx += segment.begin;
       RowIndexT ridx = d_ridx[idx];
       bst_node_t new_position = op(ridx);  // new node id
-      if (new_position == kIgnoredTreePosition) return;
```

Comment:
In `UpdatePosition()` we are back to only dealing with a single `EllpackPage` now.

Reply:
Does this mean you are abandoning external memory in the traditional sense?

Reply:
@RAMitchell @rongou By looking at the code, this two should be orthogonal.  Correct me if I'm wrong.

Reply:
@RAMitchell I guess you could say that. Streaming the whole dataset for every tree node just seems too expensive, at least with the current PCIe bandwidth/latency. If the GPU is more tightly integrated with the CPU (e.g. IBM Power9) it might be worth reconsidering.